apps/api/schedule.py

```diff
@@ -169,6 +170,56 @@ def post(self, proposal_type):
         return [t.id for t in res]
 
 
+class C3VOCPublishingWebhook(Resource):
+    method_decorators = {"post": [_require_video_api_key]}
+
+    def post(self):
+        if not request.is_json:
+            abort(415)
+
+        payload = request.get_json()
+
+        try:
+            conference = payload["fahrplan"]["conference"]
+            proposal_id = payload["fahrplan"]["id"]
+        except KeyError:
+            abort(422)
+
+        if not payload["is_master"]:
+            # c3voc *should* only send us information about the master
+            # encoding getting published. Aborting early ensures we don't
+            # accidentially delete video information from the database.
+            abort(406)
+
+        if conference != f"emf{event_year()}":
+            abort(422)
+
+        proposal = Proposal.query.get_or_404(proposal_id)
+
+        if payload["voctoweb"]["enabled"] and payload["voctoweb"]["frontend_url"]:
+            proposal.c3voc_url = payload["voctoweb"]["frontend_url"]
+            proposal.video_recording_lost = False
+        else:
+            # This allows c3voc to notify us if videos got depublished
+            # as well. We do not explicitely set 'video_recording_lost'
+            # here because the video might only need fixing audio or
+            # such.
+            proposal.c3voc_url = ""
+
+        if payload["youtube"]["enabled"] and payload["youtube"]["urls"]:
+            # c3voc will send us a list, even though we only have one
+            # video.
+            proposal.youtube_url = payload["youtube"]["urls"][0]
+            proposal.video_recording_lost = False
+        else:
+            proposal.youtube_url = ""
```

Comment:
```suggestion
        if payload["youtube"]["enabled"] and payload["youtube"]["urls"]:
            if proposal.youtube_url is None:
                # c3voc will send us a list, even though we only have one video.
                proposal.youtube_url = payload["youtube"]["urls"][0]
            else:
                # log a warning?
                pass
```

Some reasoning for this suggestion:

  * If `youtube_url` is already configured for some reason, I don't think that we should overwrite or remove the existing non-empty value.
  * Although it _sort of_ makes sense to unset `video_recording_lost` here, I think it may be better to leave it unmodified (I don't know how/why it could happen that V3VOC sends a video and yet EMF thought it was lost -- but let's allow that to be discoverable if it does?).

Reply:
@wlcx do you have opinions about unsetting `video_recording_lost` here? My assumption is "if we publish a video, the recording obviously cannot be lost", but that might be wrong?

Reply:
> If youtube_url is already configured for some reason, I don't think that we should overwrite or remove the existing non-empty value.
Agree, although you could check for equality and squash the log message if true which should help most of the time.

> do you have opinions about unsetting video_recording_lost here?
I think it's probably fine to unset it, the whole point of it was to provide a reason as to why we didn't have video URLs published - it would be very confusing to have a message saying we lost the recording and then have the recording appear below.

Reply:
> I think it's probably fine to unset it, the whole point of it was to provide a reason as to why we didn't have video URLs published - it would be very confusing to have a message saying we lost the recording and then have the recording appear below.

Ok, yep - now that I notice the [corresponding template block](https://github.com/emfcamp/Website/blob/888fb8b7d63c7560376bc2d5670b0e46f61fdce0/templates/schedule/item.html#L16-L20).. agreed, that would be confusing.  And it's probably not worthwile saying something like "we thought we'd lost it, but it's here, tada" - so unsetting the field does make sense.